### PR TITLE
cleanup zombie workers keys

### PR DESCRIPTION
**Pull request by dams**
cleanup workers keys that expired and were then recreated because increment_successful/failed_job_count or increment_total_working_time. Such workers keys end up as redis hash with only 2 fields, and *no TTL expiration* so they consume redis space for nothing

### CODE REVIEW

**Review by tchapi — APPROVED** [review]

File: rq/worker_registration.py [hunk 1]
```diff
@@ -92,10 +92,22 @@ def clean_worker_registry(queue: 'Queue'):
...
+        keys_to_check = []
 
         for i, key_exists in enumerate(results):
             if not key_exists:
                 invalid_keys.append(keys[i])
+            else:
+                keys_to_check.append(keys[i])
+
+        if keys_to_check:
+            for to_check_subset in split_list(keys_to_check, MAX_KEYS):
```
Comment (Universemul) [inline rq/worker_registration.py:104]: Do we have an idea about the time complexity of this? How often this method will be called?

Comment (dams) [reply rq/worker_registration.py:104]: it's part of the maintenance tasks, it's run every maintenance_interval, defaulting to `DEFAULT_MAINTENANCE_TASK_INTERVAL = 10 * 60`

Comment (dams) [reply rq/worker_registration.py:104]: the maintenance task already runs in `O(log(N)+M)` where N is the number of jobs in registries, M the number of jobs cleanups (times the number of job registries, but in theory irrelevant to compute run complexity)

the cleanup_workers_registry already runs in O(W) where W is the number of workers. This PR doesn't change the run complexity.

cleanup_workers_registry is part of the maintenance task, which has a running complexity of around:

```
O(
Q # constant, number of queues
*
(
      log(N)+M      # finished jobs registry
    + 2*log(N)+3*M  # started jobs registry
    + log(N)+M      # failed jobs registry
    + log(W)   # cleanup of workers
)
```

roughly equal to `O( Q*4*log(N)+5M+log(W) )`

where:
- N the number of jobs in a registry,
- M the number of jobs cleaned up in a registry
- W the workers count

**Comment by dams** [comment]
So, I thought about it:
- in the "standard usage of RQ", people tend to dedicate workers to queues. So the original code that performs cleanup of workers queue by queue is "logical". However the deletion of workers is a bit _bourrin_ because invalid workers are removed from the given queue + the global worker set.
- in our case though, all our workers are subscribed to all queues, so the code will perform the check on the same workers keys over and over again

So in the case where we want to cleanup all queues, it's better to first gather all workers keys, and clean them up at once. Not sure it's going to be a big win, given the small numbers of workers we have. So I think we're better off with the current PR code

**Pull request merged**